src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -3294,21 +3294,17 @@ void c_typecheck_baset::typecheck_function_call_arguments(
     }
     else if(i<parameter_types.size())
     {
-      const code_typet::parametert &parameter_type=
-        parameter_types[i];
+      const typet &parameter_type = parameter_types[i].type();
 
-      const typet &op_type=parameter_type.type();
-
-      if(op_type.id()==ID_bool &&
-         op.id()==ID_side_effect &&
-         op.get(ID_statement)==ID_assign &&
-         op.type().id()!=ID_bool)
+      if(
+        parameter_type.id() == ID_bool && op.id() == ID_side_effect &&
```

Comment:
:pick: If you had left `parameter_type` as being a `code_typet::parametert`, then `parameter_type.id() == ID_bool` could be `parameter_type.is_boolean()`.
:bulb: Maybe `parameter_type` should just be `parameter` and `parameter_types` should be just `parameters`. These appear to be the parameter expressions, not their types.

Reply:
Now in #7226